source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -72,7 +72,7 @@ Config::SharedConfig::SharedConfig(
     Server::Configuration::FactoryContext& context)
     : stats_scope_(context.scope().createScope(fmt::format("tcp.{}", config.stat_prefix()))),
       stats_(generateStats(*stats_scope_)),
-      flush_access_log_on_connected_(config.flush_access_log_on_connected()) {
+      flush_access_log_on_connected_(getFlushAccessLogOnConnectedConfig(config)) {
```

Comment:
nit: I think it would be better to group all these access log options initialization to one single function like `initializeAccessLogOptions`, it would make the code cleaner.

(we can remove the `const` specifier of `flush_access_log_on_connected_` then wen needn't initialize it in the initializer list.